pulsar-io/core/src/main/java/org/apache/pulsar/io/core/BatchPushSource.java

```diff
@@ -32,26 +30,43 @@
  */
 public abstract class BatchPushSource<T> implements BatchSource<T> {
 
+    private static class NullRecord implements Record {
```

Comment:
Wouldn't it be easier to allow users to place a null value inside their existing Record class and then test for that condition rather than creating a NullRecord class?  e.g.

`private static final boolean isNull(Record rec) {
	 return (rec == null) || (rec.getValue() == null);
 }`

Reply:
The thing is, is a record with a null value the same as returning null?  Can a record have a null value but have other fields e.g. key with valid values

Reply:
The Null object is currently only inserted if the record passed in is null, so there aren't any fields.

Reply:
Letting users to continue to use their own Record types and using a null value inside the record seems like a more intuitive approach to me

Reply:
That is the approach thats taken here. Users will be using their own record types and when they are done with the task, will do consume(null) to signify the end of the task. The NullRecord is strictly private class as an internal impl detail ofBatchPushSource